File: back-end/app/src/main/java/com/senescyt/app/model/Vacaciones.java
package com.senescyt.app.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.io.Serial;
import java.io.Serializable;
import java.util.Date;
import java.util.List;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name = "Vacaciones")
public class Vacaciones implements Serializable {

    @Serial
    private static final long serialVersionUID = 1L;

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "vacId")
    private Long vacId;

    /* ESTOS VAN EN EL ENCABEZADO */
    @Column(name = "vacDiasAnticipacion")
    private double vacDiasAnticipacion;
    @Column(name = "vacDiasCaducados")
    private int vacDiasCaducados;
    @Column(name = "vacDiasUsados")
    private double vacDiasUsados;
    @Column(name = "vacTotalDiasDisponibles")
    private double vacTotalDiasDisponibles;

    /* ESTOS VAN EN LA TABLA */
    @Column(name = "vacDetalle")
    private String vacDetalle;
    @Column(name = "vacDias")
    private int vacDias;
    @Column(name = "vacHoras")
    private int vacHoras;
    @Column(name = "vacMinutos")
    private int vacMinutos;
    @Column(name = "vacTotalenDias")
    private double vacTotalenDias;
    @Column(name = "vacNoGozadas")
    private double vacNoGozadas;
    @Column(name = "vacSaldo")
    private double vacSaldo;
    @Column(name = "vacFecha")
    private Date vacFecha;
    @Column(name = "vacFechaHoy")
    private Date vacFechaHoy;
    @Column(name = "vacEstado")
    private int vacEstado;
    @Column(name = "vacDiasGanados")
    private double vacDiasGanados;

    @ManyToOne
    @JoinColumn(name = "usuId", referencedColumnName = "usuId")
    private Usuario usuId;

    @JsonIgnore
    @OneToMany(mappedBy = "vacId")
    private List<Periodo> listaPeriodo;


}

File: front-end/src/app/modelo/vacaciones.ts
import { Periodos } from "./Periodos";
import { Usuario } from "./usuario";

export class Vacaciones {
    vacId: number;
    vacDetalle: string;
    vacDias: number;
    vacHoras: number;
    vacMinutos: number;
    vacSaldo: number;
    vacDiasGanados: number;
    vacNoGozadas: number;
    vacTotalenDias: number;
    vacFecha: Date;
    vacFechaHoy: Date;
    vacEstado: number;

    vacDiasAnticipacion?: number;
    vacDiasCaducados?: number;
    vacTotalDiasDisponibles?: number;
    vacDiasUsados?: number;
    periId: Periodos;
    usuId: Usuario;

    constructor(
        vacId?: number,
        vacDetalle?: string,
        vacDias?: number,
        vacHoras?: number,
        vacMinutos?: number,
        vacSaldo?: number,
        vacDiasGanados?: number,
        vacNoGozadas?: number,
        vacTotalenDias?: number,
        vacFecha?: Date,
        vacFechaHoy?: Date,
        vacEstado?: number,
        vacDiasAnticipacion?: number,
        vacDiasCaducados?: number,
        vacTotalDiasDisponibles?: number,
        vacDiasUsados?: number,
        periId?: Periodos,
        usuId?: Usuario
    ) {
        this.vacId = vacId || 0;
        this.vacDetalle = vacDetalle || '';
        this.vacDias = vacDias || 0;
        this.vacHoras = vacHoras || 0;
        this.vacMinutos = vacMinutos || 0;
        this.vacSaldo = vacSaldo || 0;
        this.vacDiasGanados = vacDiasGanados || 0;
        this.vacNoGozadas = vacNoGozadas || 0;
        this.vacTotalenDias = vacTotalenDias || 0;
        this.vacFecha = vacFecha || new Date();
        this.vacFechaHoy = vacFechaHoy || new Date();
        this.vacEstado = vacEstado || 0;
        this.vacDiasAnticipacion = vacDiasAnticipacion || 0;
        this.vacDiasCaducados = vacDiasCaducados || 0;
        this.vacTotalDiasDisponibles = vacTotalDiasDisponibles || 0;
        this.vacDiasUsados = vacDiasUsados || 0;
        this.periId = periId || new Periodos();
        this.usuId = usuId || new Usuario();
    }
}